File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/NetworkFrontDateBox.java
package com.gu.fronts.integration.test.page;

import java.util.Date;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;
import com.gu.fronts.integration.test.page.util.NetworkFrontDate;

public class NetworkFrontDateBox extends AbstractParentPage {

    @FindByTestAttribute(using = "network-front-date-title")
    private WebElement dateTitle;

    @FindByTestAttribute(using = "network-front-day-month")
    private WebElement dayOfMonth;

    public NetworkFrontDateBox(WebDriver webDriver) {
        super(webDriver);
    }

    @Override
    public NetworkFrontDateBox isDisplayed() {
        assertExists(dateTitle);
        return this;
    }

    public Date getDate() {
        return new NetworkFrontDate(dateTitle.getText()).parseToDate();
    }

    public String getDayOfWeek() {
        return new NetworkFrontDate(dateTitle.getText()).getDayOfWeek();
    }

}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/nwfront/NetworkFrontDateBox.scala
package com.gu.fronts.integration.test.page.nwfront

import java.util.Date

import org.openqa.selenium.{WebDriver, WebElement}

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.util.{FrontsParentPage, NetworkFrontDate}
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class NetworkFrontDateBox(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "network-front-date-title")
  private var dateTitle: WebElement = _

  @FindByTestAttribute(using = "network-front-day-month")
  private var dayOfMonth: WebElement = _

  override def isDisplayed(): Boolean = {
    existsAndDisplayed(dateTitle)
  }

  def getDate(): Date = {
    new NetworkFrontDate(dateTitle.getText).parseToDate()
  }

  def getDayOfWeek(): String = {
    new NetworkFrontDate(dateTitle.getText).getDayOfWeek
  }
}